viewing_party/party.py

```diff
@@ -1,23 +1,161 @@
 # ------------- WAVE 1 --------------------
 
+from re import U
+from venv import create
+
+
 def create_movie(title, genre, rating):
-    pass
+    movie = {
+        "title" : title,
+        "genre" : genre,
+        "rating" : rating,
+    }
+    if all(movie.values()) == False:
+        return None
+    else:
+        return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
 
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie):
+    for movie_dict in user_data["watchlist"]:
+        if movie_dict["title"] == movie:
+            user_data["watched"].append(user_data["watchlist"].pop())
+    return user_data
+
+      
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    x = 0.0
+    
+    if len(user_data["watched"]) == 0:
+        return x
+    
+    for film in user_data["watched"]:
+        x += film["rating"]
+    return x / len(user_data["watched"]) 

```

Comment:
Great logic! In the future, consider using a more descriptive variable name than `x`.